src/cpp/fastdds/domain/DomainParticipantImpl.cpp

```diff
@@ -1676,31 +1701,44 @@ bool DomainParticipantImpl::can_qos_be_updated(
     if (!(to.allocation() == from.allocation()))
     {
         updatable = false;
-        logWarning(RTPS_QOS_CHECK, "ParticipantResourceLimitsQos cannot be changed after the participant creation");
+        logWarning(RTPS_QOS_CHECK, "ParticipantResourceLimitsQos cannot be changed after the participant is enabled");
     }
     if (!(to.properties() == from.properties()))
     {
         updatable = false;
-        logWarning(RTPS_QOS_CHECK, "PropertyPolilyQos cannot be changed after the participant creation");
+        logWarning(RTPS_QOS_CHECK, "PropertyPolilyQos cannot be changed after the participant is enabled");
     }
     if (!(to.wire_protocol() == from.wire_protocol()))
     {
         updatable = false;
-        logWarning(RTPS_QOS_CHECK, "WireProtocolConfigQos cannot be changed after the participant creation");
+        logWarning(RTPS_QOS_CHECK, "WireProtocolConfigQos cannot be changed after the participant is enabled");
     }
     if (!(to.transport() == from.transport()))
     {
         updatable = false;
-        logWarning(RTPS_QOS_CHECK, "TransportConfigQos cannot be changed after the participant creation");
+        logWarning(RTPS_QOS_CHECK, "TransportConfigQos cannot be changed after the participant is enabled");
     }
     if (!(to.name() == from.name()))
     {
         updatable = false;
-        logWarning(RTPS_QOS_CHECK, "Participant name cannot be changed after the participant creation");
+        logWarning(RTPS_QOS_CHECK, "Participant name cannot be changed after the participant is enabled");
     }
     return updatable;
 }
 
+void DomainParticipantImpl::create_instance_handle(
+        fastrtps::rtps::InstanceHandle_t& handle)
+{
+    using eprosima::fastrtps::rtps::octet;
+
+    ++next_instance_id_;
+    handle = guid_;
+    handle.value[15] = 0x01; // Vendor specific;
+    handle.value[14] = static_cast<octet>(next_instance_id_ & 0xFF);
+    handle.value[13] = static_cast<octet>( (next_instance_id_ >> 8) & 0xFF);
```

Comment:
```suggestion
    handle.value[13] = static_cast<octet>((next_instance_id_ >> 8) & 0xFF);
```

Reply:
And, most important, on the next line we should shift by 16, not 8!